core/src/rust/filodb_core/src/query.rs

```diff
@@ -0,0 +1,997 @@
+}
+
+fn parse_boolean_query<'a>(
+    input: &'a [u8],
+    schema: &Schema,
+    default_field: Option<Field>,
+) -> IResult<&'a [u8], Box<dyn Query>, ParserError> {
+    let mut subqueries = vec![];
+    let mut next_input = input;
+    loop {
+        let (input, occur) = parse_type_id(next_input)?;
+
+        let occur = match occur {
+            TypeParseResult::Success(Occurs::Must) => tantivy::query::Occur::Must,
+            TypeParseResult::Success(Occurs::MustNot) => tantivy::query::Occur::MustNot,
+            TypeParseResult::Failure(0) => {
+                // End of boolean marker
+                next_input = input;
+                break;
+            }
+            TypeParseResult::Failure(occur) => {
+                return Err(Err::Failure(ParserError::UnknownOccur(occur)))
+            }
+        };
+
+        let (input, query) = parse_query(input, schema, default_field)?;
+
+        next_input = input;
+
+        subqueries.push((occur, query));
+    }
+
+    // Query optimization - 0 elements or 1 element can be special cased
+    let query = if subqueries.is_empty() {
+        Box::new(EmptyQuery)
+    } else if subqueries.len() == 1 && subqueries[0].0 == Occur::Must {
+        subqueries.remove(0).1
+    } else {
+        Box::new(BooleanQuery::new(subqueries))
+    };
+
+    Ok((next_input, query))
+}
+
+fn value_with_prefix(prefix: &str, value: &str) -> String {
+    format!(
+        "{}{}{}",
+        prefix,
+        if prefix.is_empty() {
+            ""
+        } else {
+            JSON_PREFIX_SEPARATOR
+        },
+        value
+    )
+}
+
+/// Build a query using a specified field and value, handling JSON
+/// fields
+///
+/// If `field` points to a valid indexed field
+/// then it is used as is and the prefix is set to an empty string
+///
+/// If `field` does not point to an indexed field then `default_field`
+/// is used and prefix is a formed JSON prefix based on `field` that
+/// should be included in any terms
+fn query_with_field_and_value<T>(
+    schema: &Schema,
+    default_field: Option<Field>,
+    field: &str,
+    func: T,
+) -> Result<Box<dyn Query>, Err<ParserError>>
+where
+    T: FnOnce(Field, &str) -> Result<Box<dyn Query>, TantivyError>,
+{
+    let Some((field, prefix)) = schema.find_field_with_default(field, default_field) else {
+        // If it's an invalid field then map to an empty query, which will emulate Lucene's behavior
+        return Ok(Box::new(EmptyQuery));
+    };
+
+    func(field, prefix).to_nom_err()
+}
+
+fn parse_equals_query<'a>(
+    input: &'a [u8],
+    schema: &Schema,
+    default_field: Option<Field>,
+) -> IResult<&'a [u8], Box<dyn Query>, ParserError> {
+    let (input, column) = parse_string(input)?;
+    let (input, text) = parse_string(input)?;
+
+    let query = query_with_field_and_value(schema, default_field, &column, |field, prefix| {
+        Ok(Box::new(TermQuery::new(
+            Term::from_field_text(field, &value_with_prefix(prefix, &text)),
```

Comment:
Wouldn't just giving the field that contains the map suffice? I found this prefix was needed only for `RegexQuery`

Reply:
The term query on a JSON field won't match if it doesn't have the correct prefix.